### ISSUE TITLE
Not Raising Errors When We Should Be?

### ISSUE THREAD

**Issue by jamesstonehill** [JsonApiClient/json_api_client#308]
It looks like we just don't raise an error when we get a response code of 400..499 except for 401, 403, 404, and 409. 
https://github.com/JsonApiClient/json_api_client/blob/master/lib/json_api_client/middleware/status.rb#L32

If so this doesn't seem right.

**Comment by gaorlov** [comment]
Good catch! That sure looks like it should raise. It may take me a week or so to get to this. Are you okay waiting, or would you be willing to spin up a PR for this?

**Comment by jamesstonehill** [comment]
@gaorlov I've fixed the omission. https://github.com/JsonApiClient/json_api_client/pull/311

Not sure if you want to include this as a minor or patch update though?

**Comment by gaorlov** [comment]
Marvelous. Thanks so much! I'll push it out tonight. Probably as `1.6.2`
